lib/src/Base/Geom/Mesh.cxx

```diff
@@ -374,10 +374,10 @@ Bool Mesh::isRegular() const
   Bool regular = true;
   const Scalar epsilon = ResourceMap::GetAsScalar("Mesh-VertexEpsilon");
   const Scalar step = vertices_(simplices_(0, 1), 0) - vertices_(simplices_(0, 0), 0);
-  for (UnsignedInteger i = 1; i < size; ++i)
+  for (UnsignedInteger i = 1; (i < size) && regular; ++ i)
   {
-    regular = regular && (std::abs(vertices_(simplices_(i, 1), 0) - vertices_(simplices_(i, 0), 0) - step) < epsilon);
-    if (!regular) break;
+    const Scalar delta = vertices_(simplices_(i, 1), 0) - vertices_(simplices_(i, 0), 0);
+    regular = regular && ((std::abs(delta - step) / std::abs(step)) < epsilon);
```

Comment:
I prefer
```
regular = (std::abs(delta - step) <= abs_step * epsilon);
```
where `abs_step` is defined outside of the loop.
Rationale:
* gracefully handle the case where step is zero
* there is no need to use boolean && operator since we exit as soon as regular is false

Reply:
Got it, thanks